File: cdk-from-cfn-testing/expected/condition_sub/csharp/ConditionSubStack.cs
using Amazon.CDK;
using Amazon.CDK.AWS.SNS;
using Constructs;
using System.Collections.Generic;

namespace ConditionSubStack
{
    public class ConditionSubStackProps : StackProps
    {
        public string EnvName { get; set; }

    }

    /// <summary>
    /// Exercises Fn::Sub inside a Condition
    /// </summary>
    public class ConditionSubStack : Stack
    {
        public ConditionSubStack(Construct scope, string id, ConditionSubStackProps props = null) : base(scope, id, props)
        {

            // Conditions
            bool hasCustomSuffix = $"{props.EnvName}-suffix" == "prod-suffix";

            // Resources
            var topic = new CfnTopic(this, "Topic", new CfnTopicProps
            {
            });
        }
    }
}

File: cdk-from-cfn-testing/expected/condition_sub/typescript/ConditionSubStack.ts
import * as cdk from 'aws-cdk-lib';
import * as sns from 'aws-cdk-lib/aws-sns';

export interface ConditionSubStackProps extends cdk.StackProps {
  /**
   */
  readonly envName: string;
}

/**
 * Exercises Fn::Sub inside a Condition
 */
export class ConditionSubStack extends cdk.Stack {
  public constructor(scope: cdk.App, id: string, props: ConditionSubStackProps) {
    super(scope, id, props);

    // Conditions
    const hasCustomSuffix = `${props.envName!}-suffix` === 'prod-suffix';

    // Resources
    const topic = hasCustomSuffix
      ? new sns.CfnTopic(this, 'Topic', {
        })
      : undefined;
    if (topic != null) {
    }
  }
}
